docs/reference/commandline/build.md

```diff
@@ -489,6 +490,49 @@ FROM alpine AS production-env
 $ docker build -t mybuildimage --target build-env .
 ```
 
+### Custom build outputs
+
+By default, a local container image is created from the build result. With the `--output` flag this behavior can be overriden and a custom exporter can be used instead. For example, this is useful for generating local binaries, code generation etc.
+
+The value for `--output` is a CSV-formatted string defining the exporter type and options. Currently, `local` and `tar` exporters are supported. The local exporter writes the resulting build files to a directory on the client side. The tar exporter is similar but writes the files as a single tarball.
+
+If no type is specified, the value defaults to the output directory of the local exporter. `-` can be used to write the output tarball to standard output.
+
+Examples:
+
+```
+docker build -o out .
+docker build -o - . > out.tar
+docker build --output type=local,dest=. .
+docker build --output type=tar,dest=out.tar .
+```
+
+A common pattern for exporting only specific files is to do multi-stage builds and to copy the desired files to a new scratch stage with [`COPY --from`](../builder.md#copy).
+
+This feature requires the use of a BuildKit backend. You can either [enable BuildKit](../builder.md#buildkit) or use the [buildx](https://github.com/docker/buildx) plugin which provides more output type options.
+
+### Specifying external cache sources
+
+In addition to local build cache, builder can reuse the cache generated from the previous builds with `--cache-from` flag pointing to an image in the registry.
+
+To use an image as a cache source, cache metadata needs to be written into the image on creation. This can be done by setting `--build-arg BUILDKIT_INLINE_CACHE=1` when building the image. After that, the built image can be used as a cache source for subsequent builds.
+
+On importing the cache, builder will only pull the JSON metadata from the registry and determine possible cache matches based on that information. If cache match is determined, the matched layers are pulled into the local environment.
+
+In addition to images, cache can also be pulled from special cache manifests generated by [buildx](https://github.com/docker/buildx) or BuildKit CLI (buildctl). Using these manifests allows to pull in layers for intermediate stages in multi-stage builds.
+
+Example:
+
+```
+docker build -t myname/myapp --build-arg BUILDKIT_INLINE_CACHE=1 .
+docker push myname/myapp
+
+# in another machine
+docker build --cache-from myname/myapp .
+```
+
+This feature requires the use of BuildKit backend. Previous builder has limited support for reusing cache from prepulled images.
```

Comment:
We might be more clear about "which feature" (I think in this case it refers to just the `BUILDKIT_INLINE_CACHE`)?

Perhaps also turn it into a note to make it stand out more

Reply:
this feature = everything in this block